internal/pkg/service/templates/api/config/config.go

```diff
@@ -1,190 +1,106 @@
-	return Repositories{
-		{
-			Type: model.RepositoryTypeGit,
-			Name: "keboola",
-			URL:  "https://github.com/keboola/keboola-as-code-templates.git",
-			Ref:  "main",
-		},
-		{
-			Type: model.RepositoryTypeGit,
-			Name: "keboola-beta",
-			URL:  "https://github.com/keboola/keboola-as-code-templates.git",
-			Ref:  "beta",
-		},
-		{
-			Type: model.RepositoryTypeGit,
-			Name: "keboola-dev",
-			URL:  "https://github.com/keboola/keboola-as-code-templates.git",
-			Ref:  "dev",
-		},
-	}
+type API struct {
+	Listen    string   `configKey:"listen" configUsage:"Listen address of the configuration HTTP API." validate:"required,hostname_port"`
+	PublicURL *url.URL `configKey:"publicURL" configUsage:"Public URL of the configuration HTTP API for link generation."  validate:"required"`
 }
 
-func NewConfig() Config {
+func New() Config {
 	return Config{
 		DebugLog:             false,
-		DebugEtcd:            false,
-		DebugHTTP:            false,
-		DatadogEnabled:       true,
-		DatadogDebug:         false,
-		CpuProfFilePath:      "",
-		ListenAddress:        "0.0.0.0:8000",
-		MetricsListenAddress: "0.0.0.0:9000",
+		DebugHTTPClient:      false,
+		CPUProfFilePath:      "",
+		NodeID:               "",
+		Datadog:              datadog.NewConfig(),
+		Etcd:                 etcdclient.NewConfig(),
+		Metrics:              prometheus.NewConfig(),
+		API:                  API{Listen: "0.0.0.0:8000", PublicURL: &url.URL{Scheme: "http", Host: "localhost:8000"}},
 		StorageAPIHost:       "",
-		Etcd: etcdclient.Config{
-			Endpoint:  "",
-			Namespace: "",
-			Username:  "",
-			Password:  "",
-		},
-		EtcdConnectTimeout:   30 * time.Second, // longer default timeout, the etcd could be started at the same time as the API
-		UniqueID:             "",
-		PublicAddress:        nil,
 		Repositories:         DefaultRepositories(),
 		TasksCleanup:         true,
 		TasksCleanupInterval: DefaultCleanupInterval,
 	}
 }
 
-func LoadFrom(args []string, envs env.Provider) (Config, error) {
-	cfg := NewConfig()
-	err := cfg.LoadFrom(args, envs)
+func Bind(args []string, envs env.Provider) (Config, error) {
+	cfg := New()
+	err := configmap.Bind(configmap.BindSpec{
+		Args:                   args,
+		EnvNaming:              env.NewNamingConvention(EnvPrefix),
+		Envs:                   envs,
+		GenerateHelpFlag:       true,
+		GenerateConfigFileFlag: true,
+		GenerateDumpConfigFlag: true,
+	}, &cfg)
 	return cfg, err
 }
 
-func (c *Config) LoadFrom(args []string, envs env.Provider) error {
-	return configmap.LoadTo(c, args, envs, EnvPrefix)
-}
-
-func (c *Config) Dump() string {
-	if kvs, err := configmap.Dump(c); err != nil {
-		panic(err)
-	} else {
-		return kvs.String()
-	}
-}
-
 func (c *Config) Normalize() {
 	c.StorageAPIHost = strhelper.NormalizeHost(c.StorageAPIHost)
-	c.Etcd.Normalize()
-	if c.PublicAddress != nil {
-		c.PublicAddress.Host = strhelper.NormalizeHost(c.PublicAddress.Host)
-		if c.PublicAddress.Scheme == "" {
```

Comment:
Odstranene normalizacie/validacie su uz definovane v inych strukturach, napr. `etcdclient.Config`:
- Aj tieto sub-struktury mozu mat metody `Normalize/Validate` (vid predchadzajuce PRs).
- Automaticky sa to vsetko validuje cez validator, takze niekde stacilo pridat `validate: ...` tag.